src/main/java/io/jenkins/plugins/pipelinegraphview/treescanner/NodeRelationship.java

```diff
@@ -0,0 +1,176 @@
+package io.jenkins.plugins.pipelinegraphview.treescanner;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.jenkins.plugins.pipelinegraphview.utils.BlueRun;
+import io.jenkins.plugins.pipelinegraphview.utils.NodeRunStatus;
+import io.jenkins.plugins.pipelinegraphview.utils.PipelineNodeUtil;
+import org.jenkinsci.plugins.workflow.actions.ErrorAction;
+import org.jenkinsci.plugins.workflow.actions.WarningAction;
+import org.jenkinsci.plugins.workflow.flow.FlowExecution;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.pipelinegraphanalysis.GenericStatus;
+import org.jenkinsci.plugins.workflow.pipelinegraphanalysis.StatusAndTiming;
+import org.jenkinsci.plugins.workflow.pipelinegraphanalysis.TimingInfo;
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException;
+import org.jenkinsci.plugins.workflow.support.actions.PauseAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NodeRelationship {
+    private boolean isDebugEnabled = logger.isDebugEnabled();
+    private static final Logger logger = LoggerFactory.getLogger(NodeRelationship.class);
+
+    @NonNull
+    protected FlowNode start;
+
+    @NonNull
+    protected FlowNode end;
+
+    @CheckForNull
+    protected FlowNode before;
+
+    @CheckForNull
+    protected FlowNode after;
+
+    public NodeRelationship(@NonNull FlowNode start, @NonNull FlowNode end, @CheckForNull FlowNode after) {
+        assert start != null;
+        assert end != null;
+        this.after = after;
+        this.start = start;
+        this.end = end;
+    }
+
+    // Print debug message if 'isDebugEnabled' is true.
+    protected void dump(String message, Object... args) {
+        if (isDebugEnabled) {
```

Comment:
🚫 I get stackoverflow error when running locally with debug enabled. I hardcoded debug to disabled and I didn't see any other issues

Reply:
I think this was a find replace issue (should be resolved already). I replaced all instances of `logger.debug` with `dump` which mean `dump` recursed on itself.